File: Exercises/Ex2/exercise2.go
package main
	
import (
	"fmt"
	"time"
	"runtime"
)




func thread1(guard chan int){
	
	
	for j:= 0; j <100000; j++{
		i:=<-guard
		i+=1
		guard <- i
	}
	
	
}

func thread2(guard chan int){
	for j:= 0; j <10000; j++{
		i:=<-guard
		i-=1
		guard <- i
	}
}

func main(){
	var i = 0
	runtime.GOMAXPROCS(runtime.NumCPU())
	guard := make(chan int,1)

	guard <- i
	go thread1(guard)
	go thread2(guard)
	
	time.Sleep(1000*time.Millisecond)
	
	fmt.Println("nubmer: ",<-guard)
}

File: Exercises/Ex2/exercise2.py
import thread,time,Queue,threading
#hallo

i = 0
#lock = threading.Lock()
queLock = Queue.Queue(1)

def main():


	global i
	thread.start_new_thread(thread1,())
	thread.start_new_thread(thread2,())
	time.sleep(10)
	print "nubmer", i


def thread1():
	#lock.acquire()
	queLock.put(1)
	global i
	for j in range(1,1000000):
		i+=1
	#lock.release()
	queLock.get()

def thread2():
	queLock.put(1)
	#lock.acquire()
	global i
	for j in range (1,1000000):
		i-=1
	#lock.release()
	queLock.get()
main()
